pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -432,4 +433,32 @@ protected void validateTopicName(String property, String namespace, String encod
             throw new RestException(Response.Status.PRECONDITION_FAILED, "Topic name is not valid");
         }
     }
+
+    protected void internalScaleTransactionCoordinators(AsyncResponse asyncResponse, int replicas) {
+        final int maxPartitions = pulsar().getConfig().getMaxNumPartitionsPerPartitionedTopic();
+        if (maxPartitions > 0 && replicas > maxPartitions) {
+            throw new RestException(Response.Status.NOT_ACCEPTABLE,
+                    "Number of partitions should be less than or equal to " + maxPartitions);
+        }
+        validateSuperUserAccessAsync().thenAccept((ignore) -> {
+                    namespaceResources().getPartitionedTopicResources()
+                            .updatePartitionedTopicAsync(SystemTopicNames.TRANSACTION_COORDINATOR_ASSIGN, p -> {
+                                if (p.partitions >= replicas) {
+                                    throw new RestException(Response.Status.NOT_ACCEPTABLE,
+                                            "Number of partitions should "
```

Comment:
```suggestion
                                            "Number of partitions should "
```
Number of transaction coordinators